lib/hitch/assets.rb

```diff
@@ -0,0 +1,84 @@
+require 'digest/md5'
+
+module Hitch
+  class Assets
+    class << self
+      def avatar_for(gravatar_email)
+        create_gravatar_images_dir! unless gravatar_images_dir_exists?
+        download_gravatar_image!(gravatar_email) unless gravatar_exists_for?(gravatar_email)
+        filename(gravatar_email)
+      end
+
+      def avatar_for_pair(email1, email2)
+        merge_avatars_for_pair!(email1, email2) unless merged_avatar_exists?(email1, email2)
+        filename(xor_email_md5s(email1, email2))
+      end
+
+      private
+
+      def xor_email_md5s(email1, email2)
+        (md5(email1).to_i(16)^md5(email2).to_i(16)).to_s(16)
+      end
+
+      def merge_avatars_for_pair!(email1, email2)
+        filename1 = File.join(gravatar_images_path, md5(email1))
+        filename2 = File.join(gravatar_images_path, md5(email2))
+        filename1xor2 = File.join(gravatar_images_path, xor_email_md5s(email1, email2))
+
+        Kernel.system("convert #{filename1} #{filename2} +append #{filename1xor2}")
```

Comment:
You're implicitly depending on ImageMagick being present on the system here.  Instead you should add MiniMagick to the Gemfile and use the ruby interface.